vendor/github.com/emicklei/go-restful/route_builder.go

```diff
@@ -113,6 +113,24 @@ func (b *RouteBuilder) Reads(sample interface{}) *RouteBuilder {
 	return b
 }
 
+// OptionalReads tells what resource type will be read from the request payload. Optional.
+// A parameter of type "body" is added ,required is set to false and the dataType is set to the qualified name of the sample's type.
+func (b *RouteBuilder) ReadsOptionalBody(sample interface{}) *RouteBuilder {
```

Comment:
We cannot change the vendor code directly, and we shouldn't push this method to upstream go-restful because it's non-standard. Please put this method in `staging/src/k8s.io/apiserver/pkg/endpoints/installer.go` and make it private e.g. 
```
func readsOptionalBody(b *RouteBuilder, sample interface{}) *RouteBuilder
```